File: Android/Java/UD3/UD03EJ012_Send/app/src/main/java/com/pmdm/ud03ej012_send/MainActivity.java
package com.pmdm.ud03ej012_send;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.google.android.material.textfield.TextInputEditText;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void mandarTexto(View view){
        TextInputEditText textoIntroducido = findViewById(R.id.TextoIntroducido);
        Intent i = new Intent(Intent.ACTION_SEND);
        i.setType("text/plain");
        i.putExtra(Intent.EXTRA_TEXT, textoIntroducido.getText().toString());
        startActivity(i);
    }

    public void llamar(View view){
        Intent i = new Intent(Intent.ACTION_DIAL, Uri.parse("tel:678989876"));
        startActivity(i);
    }
}

File: Android/Kotlin/UD3/UD03EJ012K_Send/app/src/main/java/com/pmdm/ud03ej012k_send/MainActivity.kt
package com.pmdm.ud03ej012k_send

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.textfield.TextInputEditText

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun mandarTexto(view: View?){
        var textoIntroducido: TextInputEditText = findViewById(R.id.TextoIntroducido)
        var i: Intent = Intent(Intent.ACTION_EDIT)
        i.setType("text/plain")
        i.putExtra(Intent.EXTRA_TEXT, textoIntroducido.getText().toString())
        startActivity(i)
    }

    fun llamar(view: View?){
        var i: Intent = Intent(Intent.ACTION_DIAL, Uri.parse("tel:678989876"))
        startActivity(i)
    }
}